model/client4.go

```diff
@@ -2280,256 +2300,259 @@ func (c *Client4) UploadFileAsRequestBody(data []byte, channelId string, filenam
+		return nil, BuildErrorResponse(r, appErr)
+	}
+	defer closeBody(r)
 
-		if data, err := ioutil.ReadAll(r.Body); err != nil {
-			return nil, BuildErrorResponse(r, NewAppError("GetFilePreview", "model.client.read_file.app_error", nil, err.Error(), r.StatusCode))
-		} else {
-			return data, BuildResponse(r)
-		}
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, BuildErrorResponse(r, NewAppError("GetFilePreview", "model.client.read_file.app_error", nil, err.Error(), r.StatusCode))
 	}
+	return data, BuildResponse(r)
 }
 
 // DownloadFilePreview gets the bytes for a file by id.
 func (c *Client4) DownloadFilePreview(fileId string, download bool) ([]byte, *Response) {
-	if r, err := c.DoApiGet(c.GetFileRoute(fileId)+fmt.Sprintf("/preview?download=%v", download), ""); err != nil {
-		return nil, BuildErrorResponse(r, err)
-	} else {
-		defer closeBody(r)
+	r, appErr := c.DoApiGet(c.GetFileRoute(fileId)+fmt.Sprintf("/preview?download=%v", download), "")
+	if appErr != nil {
+		return nil, BuildErrorResponse(r, appErr)
+	}
+	defer closeBody(r)
 
-		if data, err := ioutil.ReadAll(r.Body); err != nil {
-			return nil, BuildErrorResponse(r, NewAppError("DownloadFilePreview", "model.client.read_file.app_error", nil, err.Error(), r.StatusCode))
-		} else {
-			return data, BuildResponse(r)
-		}
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, BuildErrorResponse(r, NewAppError("DownloadFilePreview", "model.client.read_file.app_error", nil, err.Error(), r.StatusCode))
 	}
+	return data, BuildResponse(r)
 }
 
 // GetFileInfo gets all the file info objects.
 func (c *Client4) GetFileInfo(fileId string) (*FileInfo, *Response) {
-	if r, err := c.DoApiGet(c.GetFileRoute(fileId)+"/info", ""); err != nil {
+	r, err := c.DoApiGet(c.GetFileRoute(fileId)+"/info", "")
+	if err != nil {
 		return nil, BuildErrorResponse(r, err)
-	} else {
-		defer closeBody(r)
-		return FileInfoFromJson(r.Body), BuildResponse(r)
 	}
+	defer closeBody(r)
+	return FileInfoFromJson(r.Body), BuildResponse(r)
 }
 
 // GetFileInfosForPost gets all the file info objects attached to a post.
 func (c *Client4) GetFileInfosForPost(postId string, etag string) ([]*FileInfo, *Response) {
-	if r, err := c.DoApiGet(c.GetPostRoute(postId)+"/files/info", etag); err != nil {
+	r, err := c.DoApiGet(c.GetPostRoute(postId)+"/files/info", etag)
+	if err != nil {
 		return nil, BuildErrorResponse(r, err)
-	} else {
-		defer closeBody(r)
-		return FileInfosFromJson(r.Body), BuildResponse(r)
 	}
+	defer closeBody(r)
+	return FileInfosFromJson(r.Body), BuildResponse(r)
 }
 
 // General/System Section
 
 // GetPing will return ok if the running goRoutines are below the threshold and unhealthy for above.
 func (c *Client4) GetPing() (string, *Response) {
-	if r, err := c.DoApiGet(c.GetSystemRoute()+"/ping", ""); r != nil && r.StatusCode == 500 {
+	r, err := c.DoApiGet(c.GetSystemRoute()+"/ping", "")
+	if r != nil && r.StatusCode == 500 {
 		defer r.Body.Close()
```

Comment:
I'm not sure why this code line is here. I don't see a need to explicitly close the request body. But 0/5.